src/ir_Bosch.cpp

```diff
@@ -25,37 +27,276 @@ const uint16_t kBoschZeroSpace = 571;
+    for (uint16_t offset=0; offset < nbytes; offset += kBosch144BytesPerSection)
       // Section Header + Data + Footer
       sendGeneric(kBoschHdrMark, kBoschHdrSpace,
                   kBoschBitMark, kBoschOneSpace,
                   kBoschBitMark, kBoschZeroSpace,
                   kBoschBitMark, kBoschFooterSpace,
-                  data + offset, kSectionByteSize,
+                  data + offset, kBosch144BytesPerSection,
                   kBoschFreq, true, 0, kDutyDefault);
     space(kDefaultMessageGap);  // Complete guess
   }
 }
 
 #endif  // SEND_BOSCH144
 
+/// Class constructor.
+/// @param[in] pin GPIO to be used when sending.
+/// @param[in] inverted Is the output signal to be inverted?
+/// @param[in] use_modulation Is frequency modulation to be used?
+IRBosch144AC::IRBosch144AC(const uint16_t pin, const bool inverted,
+                           const bool use_modulation)
+    : _irsend(pin, inverted, use_modulation) { stateReset(); }
+
+/// Reset the internal state to a fixed known good state.
+void IRBosch144AC::stateReset(void) {
+  setRaw(kBosch144DefaultState);
+  setPower(true);
+}
+
+/// Set up hardware to be able to send a message.
+void IRBosch144AC::begin(void) { _irsend.begin(); }
+
+#if SEND_BOSCH144
+/// Send the current internal state as an IR message.
+/// @param[in] repeat Nr. of times the message will be repeated.
+void IRBosch144AC::send(const uint16_t repeat) {
+  if (!powerFlag) {  // "Off" is a 96bit message
+    _irsend.sendBosch144(kBosch144Off, sizeof(kBosch144Off), repeat);
+  } else {
+    _irsend.sendBosch144(getRaw(), kBosch144StateLength, repeat);
+  }   // other 96bit messages are not yet supported
+}
+#endif  // SEND_BOSCH144
+
+/// Get a copy of the internal state as a valid code for this protocol.
+/// @return A valid code for this protocol based on the current internal state.
+uint8_t* IRBosch144AC::getRaw(void) {
+  setInvertBytes();
+  setCheckSumS3();
+  return _.raw;
+}
+
+/// Set the internal state from a valid code for this protocol.
+/// @param[in] new_code A valid code for this protocol.
+void IRBosch144AC::setRaw(const uint8_t new_code[]) {
+  std::memcpy(_.raw, new_code, kBosch144StateLength);
+}
+
+void IRBosch144AC::setPower(const bool on) {
+  powerFlag = on;
+}
+
+bool IRBosch144AC::getPower(void) const {
+  return powerFlag;
+}
+
+void IRBosch144AC::setTempRaw(const uint8_t code) {
+  _.TempS1 = _.TempS2 = code >> 1;  // save 4 bits in S1 and S2
+  _.TempS3 = code & 1;              // save 1 bit in Section3
+}
+
+/// Set the temperature.
+/// @param[in] degrees The temperature in degrees celsius.
+void IRBosch144AC::setTemp(const uint8_t degrees) {
+  uint8_t temp = std::max(kBosch144TempMin, degrees);
+  temp = std::min(kBosch144TempMax, temp);
+  setTempRaw(kBosch144TempMap[temp - kBosch144TempMin]);
+}
+
+uint8_t IRBosch144AC::getTemp(void) const {
+  uint8_t temp = (_.TempS1 << 1) + _.TempS3;
+  uint8_t retemp = 0;
```

Comment:
Is a temp of `0` a good default? e.g. what happens if it can't find the temp value in the lookup table?

Reply:
changed to 25